### PR TITLE
Merge thread-utils module with internal-api

### PR DESCRIPTION
**Pull request by mcculls**
This moves the agent scheduler to the `datadog.trace.util` package in the `internal-api`

It also moves other utility classes from `internal-api` to `datadog.trace.util`

To avoid a package-prefix clash test utilities have been moved from `datadog.trace.util.test` to `datadog.trace.test.util`

### CODE REVIEW

**Review by tylerbenson — APPROVED** [review]

File: dd-java-agent/instrumentation/exception-profiling/src/test/java/com/datadog/profiling/exceptions/StreamingSamplerTest.java [hunk 1]
```diff
@@ -7,8 +7,8 @@
 import static org.mockito.ArgumentMatchers.same;
 import static org.mockito.Mockito.doNothing;
 
-import datadog.common.exec.AgentTaskScheduler;
-import datadog.common.exec.AgentTaskScheduler.Task;
+import datadog.trace.api.exec.AgentTaskScheduler;
```
[the remark this answers is not in the document]
Comment (mcculls) [reply dd-java-agent/instrumentation/exception-profiling/src/test/java/com/datadog/profiling/exceptions/StreamingSamplerTest.java:10]: I'm OK with that - there's an existing `datadog.trace.api.utils` package in `internal-api` which currently has `MathUtils` and `Strings`, should those also go in `datadog.trace.util` if we add that package?

[the remark this answers is not in the document]
Comment (tylerbenson) [reply dd-java-agent/instrumentation/exception-profiling/src/test/java/com/datadog/profiling/exceptions/StreamingSamplerTest.java:10]: yeah, I understand... I think it makes sense to have the separate package, but I don't feel strongly about it.

**Review by richardstartin — APPROVED** [review]

File: dd-java-agent/instrumentation/exception-profiling/src/test/java/com/datadog/profiling/exceptions/StreamingSamplerTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mcculls) [reply dd-java-agent/instrumentation/exception-profiling/src/test/java/com/datadog/profiling/exceptions/StreamingSamplerTest.java]: no worries, updated branch to use `datadog.trace.util`

**Review by bantonsson — APPROVED** [review]

**Review by richardstartin — APPROVED** [review]

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/WeakMapSuppliers.java [hunk 2]
```diff
@@ -1,9 +1,9 @@
 package datadog.trace.agent.tooling;
 
 import com.blogspot.mydailyjava.weaklockfree.WeakConcurrentMap;
-import datadog.common.exec.AgentTaskScheduler;
-import datadog.common.exec.AgentTaskScheduler.Task;
 import datadog.trace.bootstrap.WeakMap;
+import datadog.trace.util.AgentTaskScheduler;
```
Comment (tylerbenson) [inline dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/WeakMapSuppliers.java:5]: I would have preferred splitting up the `util` package... ie `datadog.trace.util.exec` but not a big deal.

**Pull request merged**